public/API/API_GetGameList.php

```diff
@@ -91,15 +116,19 @@
 }
 
 if ($withHashes) {
+    $responseIndex = [];
+    foreach ($response as $index => $entry) {
+        $responseIndex[$entry['ID']] = $index;
+    }
+
     $hashes = GameHash::compatible()
         ->select('game_id', 'md5')
-        ->whereIn('game_id', array_column($response, 'ID'));
+        ->whereIn('game_id', array_column($response, 'ID'))
+        ->orderBy('game_id');
+
     foreach ($hashes->get() as $hash) {
-        foreach ($response as &$entry) {
-            if ($entry['ID'] == $hash->game_id) {
-                $entry['Hashes'][] = $hash->md5;
-                break;
-            }
+        if (isset($responseIndex[$hash->game_id])) {
```

Comment:
This `isset` shouldn't be necessary as the query is limited to records in the array.